policy/modules/services/cockpit.if

```diff
@@ -0,0 +1,350 @@
+## <summary>
+##	The role template for the cockpit module.
+## </summary>
+## <desc>
+##	<p>
+##	This template creates a derived domain which is allowed
+##	to change the linux user id, to run commands as a different
+##	user.
+##	</p>
+## </desc>
+## <param name="role_prefix">
+##	<summary>
+##	The prefix of the user role (e.g., user
+##	is the prefix for user_r).
+##	</summary>
+## </param>
+## <param name="user_domain">
+##	<summary>
+##	User domain for the role.
+##	</summary>
+## </param>
+## <param name="user_exec_domain">
+##	<summary>
+##	User exec domain for execute access.
+##	</summary>
+## </param>
+## <param name="role">
+##	<summary>
+##	Role allowed access
+##	</summary>
+## </param>
+#
+template(`cockpit_role_template',`
+
+	type $1_cockpit_memfd_t;
+	files_runtime_file($1_cockpit_memfd_t)
+	dev_filetrans($2, $1_cockpit_memfd_t, file)
+
+	allow $2 $1_cockpit_memfd_t:file { manage_file_perms execute };
+
+	dev_dontaudit_execute_dev_nodes($2)
+
+	files_dontaudit_execute_default_files($2)
+	files_dontaudit_execuite_etc_runtime_files($2)
+	files_dontaudit_exec_runtime($2)
+	files_watch_etc_files($2)
+	files_watch_root_dirs($2)
+	files_watch_var_dirs($2)
+	files_watch_var_lib_dirs($2)
+
+	cockpit_use_ws_fds($2)
+	cockpit_rw_ws_stream_socket($2)
+
+	init_watch_runtime($2)
+	init_watch_utmp($2)
+
+	userdom_dontaudit_execute_user_tmpfs_files($2)
+')
+
+########################################
+## <summary>
+##	Transition to the cockpit session domain.
+## </summary>
+## <param name="domain">
+## <summary>
+##	Domain allowed to transition.
+## </summary>
+## </param>
+#
+interface(`cockpit_domtrans_session',`
+	gen_require(`
+		type cockpit_session_t, cockpit_session_exec_t;
+	')
+
+	corecmd_search_bin($1)
+	domtrans_pattern($1, cockpit_session_exec_t, cockpit_session_t)
+')
+
+########################################
+## <summary>
+##	Allow specified domain to get status of cockpit service
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`cockpit_status',`
+	gen_require(`
+		type cockpit_unit_t;
+		class service status;
+	')
+
+	allow $1 cockpit_unit_t:service status;
+')
+
+########################################
+## <summary>
+##	Allow specified domain to enable cockpit units
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`cockpit_enabledisable',`
+	gen_require(`
+		type cockpit_unit_t;
+		class service { enable disable };
+	')
+
+	allow $1 cockpit_unit_t:service { enable disable };
+')
+
+########################################
+## <summary>
+##	Allow specified domain to start cockpit units
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`cockpit_startstop',`
+	gen_require(`
+		type cockpit_unit_t;
+		class service { start stop };
+	')
+
+	allow $1 cockpit_unit_t:service { start stop };
+')
```

Comment:
`startstop` and `enabledisable` should have the `status` perm too.

Reply:
done - added use of 'cockpit_status'